### PR TITLE
(Feature) HD wallets' support

### PR DESCRIPTION
**Pull request by vbaranov**
Menu item:
screen shot 2018-11-08 at 14 13 09

Choose HD wallet page:
screen shot 2018-11-08 at 14 13 42

Trezor:
screen shot 2018-11-08 at 14 17 19

Ledger:
screen shot 2018-11-08 at 14 18 28

Guide to configure Trezor simulator: https://github.com/poanetwork/nifty-wallet/blob/master/docs/trezor-emulator.md

### CODE REVIEW

**Comment by dennis00010011b** [comment]
@vbaranov 
Improper error message if 2nd attempt to connect to Trezor

screen shot 2018-11-08 at 5 09 33 pm

Steps (don't plug-in HD wallet)
  1. Go to  'Connect to hardware wallet'
 2. Select 'Trezor', click 'Connect'
 3. Switch account or refresh the page, main screen will opens
 4.  Go to  'Connect to hardware wallet'
 5. Select 'Trezor', click 'Connect'

**Comment by dennis00010011b** [comment]
@vbaranov 
Misspelling in notification. Letter 'd' in word 'device' should be capital
screen shot 2018-11-12 at 5 45 37 pm

**Comment by dennis00010011b** [comment]
@vbaranov 
App is frozen if attempting to delete Trezor's account after choice to forget device 

Steps for reproduce:
1. Connect TrezorOne device to PC
2. Open NW, import any account from device
3. Go to Accounts-> Connect hw  and click 'Forget this device'
4. Open Accounts and select close icon for deleting Trezor's account

Actual result:
 - app unresponsive
 - uncaught errors in console [fmdpnlapjechkkicpkahdmanomngknlf-1542081145709.txt](https://github.com/poanetwork/nifty-wallet/files/2574606/fmdpnlapjechkkicpkahdmanomngknlf-1542081145709.txt)

Expected result: 
 - user should be able to delete hw account at any time 
OR
- hw accounts should be removed if user selects to forget device

screen shot 2018-11-12 at 7 52 14 pm

**Comment by vbaranov** [comment]
@dennis00010011b 

>Improper error message if 2nd attempt to connect to Trezor

Fixed here https://github.com/poanetwork/nifty-wallet/pull/183/commits/a33ba78ff1f970ba52c6e75f3ae802e89d665101

>App is frozen if attempting to delete Trezor's account after choice to forget device

Good catch! Also, fixed here https://github.com/poanetwork/nifty-wallet/pull/183/commits/a33ba78ff1f970ba52c6e75f3ae802e89d665101

>Misspelling in notification. Letter 'd' in word 'device' should be capital

Fixed  here https://github.com/poanetwork/nifty-wallet/pull/183/commits/bad4943a21dc4a2f657d8acef4b247f7df398aea

**Comment by dennis00010011b** [comment]
> > Improper error message if 2nd attempt to connect to Trezor
> 
> Fixed here [a33ba78](https://github.com/poanetwork/nifty-wallet/commit/a33ba78ff1f970ba52c6e75f3ae802e89d665101)
> 
> > App is frozen if attempting to delete Trezor's account after choice to forget device
> 
> Good catch! Also, fixed here [a33ba78](https://github.com/poanetwork/nifty-wallet/commit/a33ba78ff1f970ba52c6e75f3ae802e89d665101)
> 
> > Misspelling in notification. Letter 'd' in word 'device' should be capital
> 
> Fixed here [bad4943](https://github.com/poanetwork/nifty-wallet/commit/bad4943a21dc4a2f657d8acef4b247f7df398aea)

Tested it with Trezor One device. No issues were found

**Comment by dennis00010011b** [comment]
@vbaranov 
Uncaught error in console when try to export private key for Trezor's account
screen shot 2018-11-13 at 12 14 09 pm

**Comment by dennis00010011b** [comment]
@vbaranov 
Ledger: notification's text is incorrect when a transaction failed
screen shot 2018-11-13 at 2 23 23 pm

**Comment by dennis00010011b** [comment]
@vbaranov 
 Error messages should be more friendly and understandable for non-proficient user. Something like that:

1. Please plug in Ledger and try again
screen shot 2018-11-13 at 4 22 07 pm

2. Can not recognize device

screen shot 2018-11-13 at 4 30 08 pm

**Comment by dennis00010011b** [comment]
@vbaranov 
Ledger: list of accounts is lost (reconnection is needed ) when user exports the  private key for any account
https://www.useloom.com/share/b83694df7c544a59b2849b33d9ac4899
Steps:
 1. Connect Ledger, unlock any account
 2. Export PR for any account
 3. Go to Accounts-> Connect HW

 Expected result: list of accounts should be displayed
 Actual result: reconnection is needed

Same things happens when delete account
Device isn't really disconnected, just account list isn't displayed

**Comment by vbaranov** [comment]
@dennis00010011b 

>Ledger: notification's text is incorrect when a transaction rejected by device

This has been fixed.

Please, create separate issues for other findings. I think we could solve them in a separate PRs. They require some sort of refactoring. As about friendly errors, of course, we can parse errors from HD wallet and display friendly errors, but it will be hard to manage because errors definitions from HD device are out from our control, and if errors will be changed on HD wallet side, every time we will need to change parsing on NW side.

File: old-ui/app/components/connect-hardware/account-list.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+              onChange={(opt) => {
+                onPathChange(opt.value)
+              }}
+            />
+          </div>
+        </div>
+      )
+    }
+
+    capitalizeDevice = (device) => {
```
Comment (fvictorio) [inline old-ui/app/components/connect-hardware/account-list.js]: This function seems to do the same as `capitalizeFirstLetter`.

Comment (vbaranov) [reply old-ui/app/components/connect-hardware/account-list.js]: Thanks. Duplicate has been removed https://github.com/poanetwork/nifty-wallet/pull/183/commits/256e58527485a7622f8049d2c03f53aca058b994

**Pull request merged**